### PR TITLE
Add edit and delete role features

### PR DESCRIPTION
**Pull request by Piwakk**
Some parts in `EditRoleModal.tsx` are quite redundant / heavy.
I will be able to simplify them after the refactoring of the date inputs, which will come in a further PR

### CODE REVIEW

File: frontend/src/components/associations/roles/administration/EditRoleModal.tsx [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+import { TextFormGroup } from "../../../utils/forms/TextFormGroup";
+import { DayPickerInputFormGroup } from "../../../utils/forms/DayPickerInputFormGroup";
+import { SwitchCheckbox } from "../../../utils/forms/SwitchCheckbox";
+import { SelectGroup } from "../../../utils/forms/SelectGroup";
+import { RolePermissionIcon } from "./RolePermissionIcon";
+import { RolePermissionTooltip } from "./RolePermissionTooltip";
+import "./edit_role_modal.css";
+
+const permissionItems = new Map([
+    [
```
Comment (abocquet) [inline frontend/src/components/associations/roles/administration/EditRoleModal.tsx:20]: Use `["association", "administration, ...].map(type => [type,  <Element type={type} />])`

File: frontend/src/components/associations/roles/administration/EditRoleModal.tsx [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+                          (error.response.status === 403
+                              ? "vous n’avez pas le droit de modifier ce sondage."
+                              : error.response.data.detail)
+                        : ""
+                }`
+            );
+        },
+    });
+
+    return role !== null ? (
```
Comment (abocquet) [inline frontend/src/components/associations/roles/administration/EditRoleModal.tsx:109]: Same comment as before, return the null case before

File: frontend/src/components/user/profile/ProfileAssociations.tsx [hunk 3]
```diff
@@ -50,7 +50,11 @@ export const ProfileAssociations = ({ profile }: { profile: Profile }) => (
                                     </span>
                                 }
                                 startDate={role.startDate}
-                                endDate={role.endDate}
+                                endDate={
+                                    role.endDate === null
+                                        ? undefined
+                                        : role.endDate
```
Comment (abocquet) [inline frontend/src/components/user/profile/ProfileAssociations.tsx:56]:
```suggestion
                                    role.endDate || undefined
```

File: frontend/src/components/utils/forms/DayPickerInputField.tsx [hunk 4]
```diff
@@ -83,13 +90,15 @@ export const DayPickerInputField = ({
                     ...dayPickerLocalisationProps(todayButton),
                 }}
                 inputProps={{
-                    className: `form-control ${
-                        feedback && meta.error ? "is-invalid" : ""
+                    className: `${inputProps ? inputProps : ""} form-control ${
```
Comment (abocquet) [inline frontend/src/components/utils/forms/DayPickerInputField.tsx:93]:
```suggestion
                    className: `${inputProps || ""} form-control ${
```

File: backend/associations/serializers/role.py [hunk 5]
```diff
@@ -22,17 +37,34 @@ def __init__(self, used_by_association_admin=False, *args, **kwargs):
...
 
     def save(self, **kwargs):
-        """If the user is not an association admin, only give them access to the administration_permission field."""
-        if not self.used_by_association_admin:
-            forbidden_fields = tuple(
-                f"{permission_name}_permission"
-                for permission_name in Role.PERMISSION_NAMES
-                if permission_name != "administration"
-            ) + ("start_date", "end_date", "role", "rank")
+        """If the user is not an association admin, only give them access to the "administration" permission."""
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply backend/associations/serializers/role.py:40]: Let me restate it:
* On any given association, only the association admins are allowed to edit every permission.
* However, a global admin can edit the `administration` permission.
* So, when the comment says “if the user is not an association admin”, it is implying that the user is a global admin but not an association admin. This check is taken care of by the view.

`permissions` is a property, very convenient for returning the permissions without having to check them all one by one, and also for changing them, even if this requires a little bit of customization work.

File: frontend/src/components/associations/roles/administration/AssociationRolesAdministration.tsx [hunk 6]
```diff
@@ -106,16 +128,36 @@ export const AssociationRolesAdministration = ({
...
+        onSettled: () => queryCache.refetchQueries("roles.list"),
+    });
 
-    const { columns, sorting } = useColumns<Role>(columnData(setEditRole));
+    const { columns, sorting } = useColumns<Role>(
+        columnData(setEditRole, remove)
+    );
 
-    return (
+    return association.myRole?.permissions?.includes("administration") ? (
```
[the remark this answers is not in the document]
Comment (Piwakk) [reply frontend/src/components/associations/roles/administration/AssociationRolesAdministration.tsx:151]: I did it, but with Prettier rules, this actually does not change the indentation

**Comment by abocquet** [comment]
The sun shines and this PR is approved

**Pull request merged**